src/common/Page.ts

```diff
@@ -1362,10 +1362,17 @@ export class Page extends EventEmitter {
 
   /**
    * @param userAgent - Specific user agent to use in this page
+   * @param userAgentData - Specific user agent client hint data to use in this
+   * page
    * @returns Promise which resolves when the user agent is set.
    */
-  async setUserAgent(userAgent: string): Promise<void> {
-    return this._frameManager.networkManager().setUserAgent(userAgent);
+  async setUserAgent(
```

Comment:
we'd need to update the tests in https://github.com/puppeteer/puppeteer/blob/main/test/page.spec.ts#L962

Reply:
I swear I searched for tests and just failed to find them… but this is ideal, I'll update these to include cases for the new parameter.